contracts/Multisig.vy

```diff
@@ -0,0 +1,150 @@
+# STATE VARIABLES
+#######################################################################################  
+owners: DynArray[address, 100]
+isOwner: HashMap[address, bool]
+numConfirmationsRequired: uint256
+
+# Mapping from txIndex to a mapping of [address, bool], denoting the tx confirmation by a given address
+isConfirmed: public(HashMap[uint256, HashMap[address, bool]])
+
+# Dynamic Array with all the submitted transactions
+# Can we increase this to (2 ^ 256) - 1? 
+transactions: DynArray[Transaction, 10000]
+
+#######################################################################################  
+# INITIALIZATION FUNCTION
+#######################################################################################  
+@external
+def __init__ (_numConfirmationsRequired: uint256, _owners: DynArray[address,100]):
+    assert len(_owners) > 0, "Error: No Owners"
+    assert _numConfirmationsRequired > 0 and _numConfirmationsRequired <= len(_owners)
+    self.numConfirmationsRequired = _numConfirmationsRequired
+    for o in _owners:
+        assert o != empty(address)
+        assert self.isOwner[o] == False, "Owner Not Unique"
+        self.isOwner[o] = True
+        self.owners.append(o)
+
+#######################################################################################  
+# EXTERNAL FUNCTIONS
+####################################################################################### 
+@external
+@payable
+def __default__():
+    log Deposit(msg.sender, msg.value, self.balance)
+
+@external
+def submitTransaction( _to: address, _val: uint256, _data: Bytes[256]):
+    assert self.isOwner[msg.sender] == True, "Not an Owner"
+    txIndex: uint256 = len(self.transactions)
+    self.transactions.append(Transaction({to:_to, val:_val, data: _data, executed: False, numConfirmations: 0}))    
+
+@external 
+def confirmTransaction(txIndex: uint256):
+    assert self.isOwner[msg.sender]==True, "Not an Owner"
+    assert txIndex < len(self.transactions), "Does Not Exist"
+    assert self.transactions[txIndex].executed == False, "Transaction Already Executed"
+    assert self.isConfirmed[txIndex][msg.sender] == False, "Transaction Already Confirmed"
+    # increment the number of tx confirmations
+    self.transactions[txIndex].numConfirmations += 1
+    # take note of the msg sender signing the tx
+    self.isConfirmed[txIndex][msg.sender] = True
+    log ConfirmTransaction(msg.sender, txIndex)
+
+
+@payable
+@external
+def executeTransaction(txIndex: uint256) -> Bytes[256]:
+    assert self.isOwner[msg.sender]==True, "Not an Owner"
+    assert txIndex < len(self.transactions), "Does Not Exist"
+    assert self.transactions[txIndex].executed == False, "Transaction Already Executed"
+    assert self.transactions[txIndex].numConfirmations >= self.numConfirmationsRequired, "Insufficent Confirmations"
+    assert self.transactions[txIndex].val <= self.balance, "Insufficient multisig balance"

```

Comment:
Unnecessary check: transaction reverts if there is an attempt to spend an amount > balance.